client/components/workouts/LogWorkout.jsx

```diff
@@ -0,0 +1,278 @@
+
+
+export default function LogWorkout() {
+    const [selectedDate, setSelectedDate] = useState(new Date());
+    const [loading, setLoading] = useState(false);
+    const navigate = useNavigate();
+
+    const [workouts, setWorkouts] = useState({
+        cardio: [],
+        strength: [],
+        flexibility: [],
+        sports: []
+    });
+
+    const dailyGoals = {
+        calories: 2800,
+        duration: 60, // will change depending on their onboarding choice
+        exercises: 5,
+    };
+
+    const calculateTotals = () => {
+        let totals = { calories: 0, duration: 0, exercise: 0 };
+
+        Object.values(workouts).flat().forEach(workout => {
+            totals.calories += workout.calories || 0;
+            totals.duration += workout.duration || 0;
+            totals.exercises += 1;
+        });
+        return totals;
+    };
+
+
+    const totals = calculateTotals();
+    const remaining = {
+        calories: Math.max(0, dailyGoals.calories - totals.calories),
+        duration: Math.max(0, dailyGoals.duration - totals.duration),
+        exercises: Math.max(0, dailyGoals.exercises - totals.exercises),
+    };
+
+    const formatDate = (date) => {
+        return date.toLocaleDateString("en-US", {
+            weekday: "long",
+            month: "long",
+            day: "numeric",
+            year: "numeric"
+        });
+    };
+
+    const handleAddWorkout = (workoutType) => {
+        console.log("add workout to", workoutType);
+        // TODO: Add workout modal functionality
+    };
+
+
+    const handleQuickTools = (workoutType) => {
+        console.log("quick tools for", workoutType);
+        // TODO: Add quick tools functionality
+    };
+
+
+    const handleRemoveWorkout = (workoutType, idx) => {
+        console.log("remove workout from", workoutType, "at index", idx);
+        // TODO: Add remove workout functionality
+    };
+
+
+    if (loading) {
+        return (
+        <div className="log-workout-page">
+            <button
+            className="back-btn loading"
+            onClick={() => navigate("/dashboard")}
+            >
+            Back to Dashboard
+            </button>
+            <div className="log-workout-container">
+            <div className="loading-state">
+                <p>Loading your workouts...</p>
+            </div>
+            </div>
+        </div>
+        );
+    }
+
+
+    return (
+        <div className="log-workout-page">
+        <button className="back-btn" onClick={() => navigate("/dashboard")}>
+            Back to Dashboard
+        </button>
+
+
+        <div className="log-workout-container">
+            <header className="log-workout-header">
+            <h1>Your workout log for:</h1>
+            <div className="date-selector">
+                <button
+                className="date-nav-btn"
+                onClick={() =>
+                    setSelectedDate(
+                    new Date(selectedDate.getTime() - 24 * 60 * 60 * 1000)
```

Comment:
Please move 24 * 60 * 60 * 1000 to a const
```suggestion
                    new Date(selectedDate.getTime() - ONE_DAY_MS)
```